integrations/docs/API/swagger.yaml

```diff
@@ -0,0 +1,287 @@
+      requestBody:
+        required: true
+        description: Inventory item to add
+        content:
+          application/json:
+            schema:
+              $ref: '../../../schema/system/integration.schema'
+      responses:
+        '201':
+          description: item created
+        '400':
+          description: 'invalid input, object invalid'
+        '409':
+          description: an existing item already exists
+  /store:
+    get:
+      tags:
+        - developers
+      summary: get all stored Integrations
+      operationId: searchStore
+      description: |
+        By passing in the appropriate search attributes, you can search for
+        available Integrations in the store
+      parameters:
+        - in: query
+          name: searchString
+          description: pass an optional search string for looking up Integration in the store
+          required: false
+          schema:
+            type: string
+        - in: query
+          name: limit
+          description: maximum number of records to return
+          schema:
+            type: integer
+            format: int32
+            minimum: 0
+            maximum: 50
+      responses:
+        '200':
+          description: search results matching criteria
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '../../../schema/system/integration-instance.schema'
+        '400':
+          description: bad input parameter
+    post:
+      tags:
+        - admins
+      summary: adds an Integration item to store
+      operationId: storeIntegration
+      description: |
+       Adds an Integration to the system store, expecting the internal URLs to be accessible
+      requestBody:
+        required: true
+        description: Inventory item to add
+        content:
+          application/json:
+            schema:
+              $ref: '../../../schema/system/integration-instance.schema'
+      responses:
+        '201':
+          description: item created
+        '400':
+          description: 'invalid input, object invalid'
+        '409':
+          description: an existing item already exists
+  /store/{id}/status:
+    get:
+      tags:
+        - developers
+      summary: get status for a loaded Integration
+      operationId: integrationStatus
+      description: |
+        Get the Integration's status in the store
+      parameters:
+        - in: path
+          name: id
+          description: Integration instance Id
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: integration store status
+          content:
+            application/json:
+              schema:
+                $ref: '../../../schema/system/integration-instance.schema'
+        '400':
+          description: bad input parameter
+  /store/{id}/validate:
```

Comment:
Is this swagger for both internal/external APIs? This seems like it should be an internal API called only after user tries to store something?

Reply:
I would also expect validation to be a part of the storage routine. My understanding is that it's because an integration has many referential parts and you can't do a full validation before every part is loaded. But where are we getting circular references that causes that?